### ISSUE TITLE
Exits while parsing legends_plus file, leaves browser stuck on "loading" page

### ISSUE THREAD

**Issue by myk002** [robertjanetzko/LegendsBrowser2#18]
Console output:
```
Loading: /home/myk/.steam/steam/steamapps/common/Dwarf Fortress/region4-00005-01-01-legends.xml
278.53 KiB / 278.53 KiB [------------------------------------------------------------------------------------------] 100.00% ? p/s 0s

Loading: /home/myk/.steam/steam/steamapps/common/Dwarf Fortress/region4-00005-01-01-legends_plus.xml
2024/02/15 08:34:11 EOF
```

Video:

https://github.com/robertjanetzko/LegendsBrowser2/assets/977482/af4aa066-3b78-4a8e-83ec-a4d567ea6867

**Comment by myk002** [comment]
legends XML files:
[legends_xml.zip](https://github.com/robertjanetzko/LegendsBrowser2/files/14299576/legends_xml.zip)

**Comment by robertjanetzko** [comment]
Thank you for your feedback, but the issue seems to be with the dfhack export. The file structure isn't correct:

```
<?xml version="1.0" encoding='UTF-8'?>
<df_world>
<name>Nitom Arkoth</name>
<altname>The Planets of Prophecy</altname>
<landmasses>
	<landmass>
		<id>0</id>
		<name>The Land of Constructs</name>
		<coord_1>1,1</coord_1>
		<coord_2>15,15</coord_2>
	</landmass>
</landmasses>
<landmasses>
	<mountain_peak>
		<id>0</id>
		<name>The Deteriorated King</name>
		<coords>8,6</coords>
		<height>270</height>
	</mountain_peak>
	<mountain_peak>
		<id>1</id>
		<name>The Silken Blazes</name>
		<coords>9,15</coords>
		<height>177</height>
		<is_volcano/>
	</mountain_peak>
</landmasses>
<landmasses>
	<region>
		<id>0</id>
		<coords>0,0|0,1|0,2|0,3|0,4|0,5|0,6|0,7|0,8|0,9|0,10|0,11|0,12|0,13|0,14|0,15|0,16|1,0|1,1|1,2|1,3|1,4|1,6|1,7|1,9|1,10|1,11|1,12|1,13|1,14|1,15|1,16|2,0|2,1|2,2|2,9|2,10|2,11|2,12|2,13|2,14|2,15|2,16|3,0|3,1|3,2|3,3|3,10|3,11|3,13|3,15|3,16|4,0|4,1|4,15|4,16|5,0|5,1|5,15|5,16|6,0|6,16|7,0|7,16|8,0|8,15|8,16|9,0|9,16|10,0|10,16|11,0|11,16|12,0|12,13|12,16|13,0|13,1|13,2|13,4|13,12|13,13|13,14|13,15|13,16|14,0|14,1|14,2|14,3|14,4|14,5|14,6|14,11|14,12|14,13|14,14|14,15|14,16|15,0|15,1|15,2|15,3|15,4|15,5|15,6|15,9|15,10|15,12|15,13|15,14|15,15|15,16|16,0|16,1|16,2|16,3|16,4|16,5|16,6|16,7|16,8|16,9|16,10|16,11|16,12|16,13|16,14|16,15|16,16|</coords>
		<evilness>neutral</evilness>
	</region>
```

As you can see each section starts with `landmasses` instead of `mountain_peaks` and `regions`. Perhaps you can report this issue in the dfhack repo?

**Comment by im-coooding** [comment]
So I reported this in the dfhack repo, but I see now that it's your commit @myk002 that caused this issue in the first place lmao.

No disrespect intended, you do very good work, I just found that a bit funny.

**Comment by myk002** [comment]
It was an amusingly circuitous route. I made the coding error when I modified `exportlegends` to have a smoothly advancing progress bar across the export, then I saw a report on reddit about LegendsBrowser crashing. I didn't connect it to my recent change (it hasn't been released in a stable release yet). I thought this was related to other structure names that have been changing recently, such as [this one](http://www.bay12forums.com/smf/index.php?topic=179848.msg8521966#msg8521966). Of course it all comes back to a facepalm.

Anyway, thank you for LegendsBrowser2 -- the whole community really loves it and appreciates it!
